README.md

```diff
@@ -202,6 +210,123 @@ with pool.connect() as db_conn:
         print(row)
 ```
 
+### Async Driver Usage
+
+The AlloyDB Connector is compatible with [asyncio][] to improve the speed and
+efficiency of database connections through concurrency. You can use all
+non-asyncio drivers through the `Connector.connect_async` function, in addition
+to the following asyncio database drivers:
+
+- [asyncpg](https://magicstack.github.io/asyncpg)
+
+[asyncio]: https://docs.python.org/3/library/asyncio.html
+
+```python
+import asyncpg
+
+import sqlalchemy
+from sqlalchemy.ext.asyncio import AsyncEngine, create_async_engine
+
+from google.cloud.alloydb.connector import Connector
+
+async def init_connection_pool(connector: Connector) -> AsyncEngine:
+    # initialize Connector object for connections to AlloyDB
+    async def getconn() -> asyncpg.Connection:
+        conn: asyncpg.Connection = await connector.connect_async(
+            "projects/<YOUR_PROJECT>/locations/<YOUR_REGION>/clusters/<YOUR_CLUSTER>/instances/<YOUR_INSTANCE>",
+            "asyncpg",
+            user="my-user",
+            password="my-password",
+            db="my-db-name"
+            # ... additional database driver args
+        )
+        return conn
+
+    # The AlloyDB Python Connector can be used along with SQLAlchemy using the
+    # 'async_creator' argument to 'create_async_engine'
+    pool = create_async_engine(
+        "postgresql+asyncpg://",
+        async_creator=getconn,
+    )
+    return pool
+
+async def main():
+    loop = asyncio.get_running_loop()
+    connector = Connector(loop=loop)
+
+    # initialize connection pool
+    pool = await init_connection_pool(connector)
+
+    # example query
+    async with pool.connect() as conn:
+        await conn.execute(sqlalchemy.text("SELECT NOW()"))
+
+    # close Connector
+    await connector.close_async()
+
+    # dispose of connection pool
+    await pool.dispose()
```

Comment:
nit: should close in LIFO order to avoid issues
```suggestion
    # dispose of connection pool
    await pool.dispose()
    
    # close Connector
    await connector.close_async()
```